### ISSUE TITLE
Renaming the `systems` directory

### ISSUE THREAD

**Issue by digikar99** [ocicl/ocicl#60]
Hello, it's been great using ocicl. Thank you for developing it!

I wanted to know if it's possible to rename the `systems` directory to something else without touching the source code. So, I'm looking for a configuration option / global variable to do this. I develop with multiple languages, and I'd prefer that directory to be called `ocicl-systems` or something more specific.

**Comment by atgreen** [comment]
Thanks for the feedback!  I think an option you configure at 'ocicl setup' time would be fine.  I am not able to implement this any time soon, but would be happy to look at any patches you may come up with.

**Comment by digikar99** [comment]
I made some local changes - amounting to replacing all instance of `(:relative "systems")` with `(:relative "ocicl-system")`. And also another similar change in the `runtime` directory. But this isn't the solution for the problem.

I'll issue a PR if I end up adding a proper configuration option!

**Comment by dieggsy** [comment]
I actually think this would be particularly useful non-globally, e.g. something like

`ocicl install --systems-dir=./my-deps`

**Comment by dieggsy** [comment]
Two thoughts:

- If `systems` gets renamed, `systems.csv` probably should too
- It would be great if the ocicl standard default was more specific than `systems`, which is not particularly unique and has a higher chance of name clashing than something containing `ocicl` in the name.

**Comment by dieggsy** [comment]
One potential issue I see with allowing a user-configurable systems name is fragmentation. If everyone uses their preferred name for systems/system.csv, it becomes kind of harder to use ocicl in a unified way across projects without every user adding the systems aliases of the projects they build. Maybe this is a non-issue in practice if "most people" stick to the default, but I think it's worth considering.

I wonder if there's some way to have both (people can call systems what they want, ocicl knows where to look automatically). Maybe config files? Maybe systems.csv could instead be called <whatever>.ocicl, instructing ocicl to then create <whatever> as the systems directory name.

All that said, I do still think the default should include `ocicl` in the name. If any changes to the default were made, it would be nice if they were done in a backward-compatible way (e.g. look for ocicl-systems first, then systems, optionally warn of deprecation).

**Comment by digikar99** [comment]
> One potential issue I see with allowing a user-configurable systems name is fragmentation. If everyone uses their preferred name for systems/system.csv, it becomes kind of harder to use ocicl in a unified way across projects without every user adding the systems aliases of the projects they build. Maybe this is a non-issue in practice if "most people" stick to the default, but I think it's worth considering.

That's a good point. Imagine navigating to a new project and not knowing which of the several csv files to look up.

I think that makes me want to have only a global configuration option and not local ones.

> All that said, I do still think the default should include ocicl in the name. If any changes to the default were made, it would be nice if they were done in a backward-compatible way (e.g. look for ocicl-systems first, then systems, optionally warn of deprecation).

I do agree with including ocicl in the name for the default (non-configurable?) names of the local. How about the naive ocicl-systems.csv/ocicl-systems?

**Comment by atgreen** [comment]
Having given this some thought, I agree with @digikar99 that ocicl-systems.csv / ocicl-systems is probably best (allowing  systems.csv / systems for backwards compatibility).  I don't think we need to support any more flexibility than that.  I strongly prefer keeping a .csv extension so that csv-reading program (spreadsheet programs, etc) will know what to do with it.

**Comment by dieggsy** [comment]
While I do think the implemented PR is rather "cool", I would be willing to give it up for the sake of simplicity. The point about csv-reading programs that rely on the extension is very fair.

My only very minor, almost aesthetic gripe with ocicl-systems is it's a little more verbose and single word + extension feels neater. I recognize that's a little petty, and I can't think of a better alternative since ocicl-systems is pretty descriptive and unambiguous. Plus, I can't be against promoting lisp-case!

I can reimplement using @digikar99's proposal unless further input changes course

**Comment by digikar99** [comment]
> My only very minor, almost aesthetic gripe with ocicl-systems is it's a little more verbose and single word + extension feels neater.

I agree! At the same time, I want to include csv as the extension. May be just ocicl.csv/ocicl? But really, I'm bad at naming.

**Comment by dieggsy** [comment]
I wouldn't mind ocicl.csv / ocicl either

**Comment by dieggsy** [comment]
@atgreen any thoughts on ocicl.csv / ocicl as more compact alternatives?
